### DISCUSSION TITLE
Add the possibility to change the rendered font arbitrarily

### DISCUSSION THREAD

**Discussion by LeonardoCT** [sumatrapdfreader/sumatrapdf#3439] [Ideas]
Hello, I have dyslexia, and it would be very cool if you could arbitrarily change the rendered font to a custom one like open dyslexic, or others, I'm aware that doing this might result in bad formation of the pdf contents, but right now there are no options (paid or free) to do such things, and it would be cool if sumatra pdf could implement this feature

**Comment by kjk** [comment]
Well, there is no such option because it's pretty much impossible.

PDF format specifies an exact position of each letter. If you change that letter to use a different font then its size will change and everything will get messed up.

**Reply by GitHubRulesOK** [reply]
@LeonardoCT 
To rewrite areas as a different font you need an editor that can supply the new font to replace old, HOWEVER not all text is replaceable for MANY reasons
- Text may be pixels thus need a total replacement that are fresh visual images of each character
- Text may be simple pen strokes that in effect would need conversion into characters then those characters replaced with a font
- Text may in a PDF not be what's shown as surface ink, so its not uncommon when copy and paste that text you get is rubbish (we call that "MojiBake"). Sometimes that is correct ordering but rotated numerically (i.e. recoverable), sometimes they are exceptionally random thus almost impossible to decipher.
- Text will be only written in a set of co-ordinates that are targeting one line in dribs and drabs so altering letters will scroll the end of line out of sight
- Text width is King so if its 10 points font it may be 7 points high visually,  replace with another 10 point wide font and the lines could overlap in height and width

Basically the reason you cant find an app is you need to copy all that pages text and paste in a new font say in NotePad Pages and then apply some form of design layout to suite the new font widths. Finally convert the new text into a PDF possibly with a different number of pages.
